daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -30,6 +30,13 @@ private[lf] object Compiler {
   case class PackageNotFound(pkgId: PackageId)
       extends RuntimeException(s"Package not found $pkgId", null, true, false)
 
+  // NOTE(MH): We make this an enum type to avoid boolean blindness. In fact,
+  // other profiling modes like "only trace the ledger interactions" might also
+  // be useful.
+  sealed abstract class ProfilingMode
```

Comment:
```suggestion
  sealed abstract class ProfilingMode extends Product with Serializable
```

In general for ADT roots when all subclasses are `case`s.

Reply:
What does this do and what's the benefit of doing it?

Reply:
Same as scala/bug#9173, with the fix being scala/scala#4355 in that case.

It also eliminates false positives on the `Product` and `Serializable` warts.